docs/content/architecture/high-availability.md

```diff
@@ -60,79 +60,18 @@ the availability of any of the PostgreSQL clusters that it manages, as the
-not be immediately replayed. We will address how we handle this further down in
-this section.
-
-During this process, each replica keeps track of how far along in the recovery
-process it is using a "log sequence number" (LSN), a built-in PostgreSQL serial
-representation of how many logs have been replayed on each replica. For the
-purposes of HA, there are two LSNs that need to be considered: the LSN for the
-last log received by the replica, and the LSN for the changes replayed for the
-replica. The LSN for the latest changes received can be compared amongst the
-replicas to determine which one has replayed the most changes, and an important
-part of the automated failover process.
-
-The replicas periodically check in on the lock to see if it has been updated by
-the primary within the allowable automated failover timeout. Each replica checks
-in at a randomly set interval, which is a key part of Raft algorithm that helps
-to ensure consensus during an election process. If a replica believes that the
-primary is unavailable, it becomes a candidate and initiates an election and
-votes for itself as the new primary. A candidate must receive a majority of
-votes in a cluster in order to be elected as the new primary.
-
-There are several cases for how the election can occur. If a replica believes
-that a primary is down and starts an election, but the primary is actually not
-down, the replica will not receive enough votes to become a new primary and will
-go back to following and replaying the changes from the primary.
-
-In the case where the primary is down, the first replica to notice this starts
-an election. Per the Raft algorithm, each available replica compares which one
-has the latest changes available, based upon the LSN of the latest logs
-received. The replica with the latest LSN wins and receives the vote of the
-other replica. The replica with the majority of the votes wins. In the event
-that two replicas' logs have the same LSN, the tie goes to the replica that
-initiated the voting request.
-
-Once an election is decided, the winning replica is immediately promoted to be a
-primary and takes a new lock in the distributed etcd cluster. If the new primary
-has not finished replaying all of its transactions logs, it must do so in order
-to reach the desired state based on the LSN. Once the logs are finished being
-replayed, the primary is able to accept new queries.
-
-At this point, any existing replicas are updated to follow the new primary.
-
-When the old primary tries to become available again, it realizes that it has
-been deposed as the leader and must be healed. The old primary determines what
-kind of replica it should be based upon the CRD, which allows it to set itself
-up with appropriate attributes.  It is then restored from the pgBackRest backup
-archive using the "delta restore" feature, which heals the instance and makes it
-ready to follow the new primary, which is known as "auto healing."
```

Comment:
👍🏻

🌱 I can imagine some value in describing what happens to the "old primary," but with less detail than is here, maybe.